### PR TITLE
add basic models

### PR DESCRIPTION
**Pull request by bits0rcerer**
Add models based on our design document.

### CODE REVIEW

**Review by Wil-Lee — COMMENTED** [review]
looks fine

**Review by Wil-Lee — APPROVED** [review]

**Review by Wil-Lee — APPROVED** [review]

File: src/main/java/com/peerrequest/app/model/BiddingSlot.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+     * Ratings of a bidding slot.
+     * May be null.
+     */
+    @Getter
+    @Setter
+    private List<Rating> ratings;
+
+    @Getter
+    @Setter
+    private int reviewSlots = 1;
```
Comment (ToastOfToastyness) [inline src/main/java/com/peerrequest/app/model/BiddingSlot.java:50]: Why is it initalized with 1?

File: src/main/java/com/peerrequest/app/model/BiddingSlot.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+import java.util.List;
+
+/**
+ * This class represents a bidding slot.
+ *
+ * @author Michael Mildt
+ * @version 0.0.1
+ */
+public class BiddingSlot {
```
Comment (ToastOfToastyness) [inline src/main/java/com/peerrequest/app/model/BiddingSlot.java:14]: methods addRating and removeRating are missing

File: src/main/java/com/peerrequest/app/model/Request.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import lombok.Getter;
+
+/**
+ * This abstract class represents a request.
+ *
+ * @author Nikola Boban
+ * @author Michael Mildt
+ * @version 0.0.1
+ */
+public abstract class Request {
```
Comment (ToastOfToastyness) [inline src/main/java/com/peerrequest/app/model/Request.java:13]: Attribute requestState and methode setState are missing

File: src/main/java/com/peerrequest/app/model/Review.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+import lombok.Setter;
+
+/**
+ * This class represents a review.
+ *
+ * @author Luder Halpick
+ * @author Michael Mildt
+ * @version 0.0.1
+ */
+public class Review {
```
Comment (ToastOfToastyness) [inline src/main/java/com/peerrequest/app/model/Review.java:14]: Relation to Message,java missing

File: src/main/java/com/peerrequest/app/model/Category.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+
+/**
+ * This class represents a category.
+ *
+ * @author Ludger Halpick
+ * @author Michael Mildt
+ * @version 0.0.1
+ */
+public class Category {
```
Comment (ToastOfToastyness) [inline src/main/java/com/peerrequest/app/model/Category.java:17]: relation to Entry.java missing

File: src/main/java/com/peerrequest/app/model/Entry.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+import lombok.Setter;
+
+/**
+ * This class represents an entry.
+ *
+ * @author Ludger Halpick
+ * @author Michael Mildt
+ * @version 0.0.1
+ */
+public class Entry {
```
Comment (ToastOfToastyness) [inline src/main/java/com/peerrequest/app/model/Entry.java:14]: Relation to Category.java and Review.java missing

File: src/main/java/com/peerrequest/app/model/BiddingSlot.java (hunk 1 above)
Comment (Wil-Lee) [reply src/main/java/com/peerrequest/app/model/BiddingSlot.java:50]: Default Value specified in Requirement Specification

File: src/main/java/com/peerrequest/app/model/Entry.java (hunk 6 above)
Comment (Wil-Lee) [reply src/main/java/com/peerrequest/app/model/Entry.java:14]: same goes for review relationship

File: src/main/java/com/peerrequest/app/model/Category.java (hunk 5 above)
Comment (Wil-Lee) [reply src/main/java/com/peerrequest/app/model/Category.java:17]: Is not needed . All related entries are identified by their category attribute

**Review by okidokidokkaebi — APPROVED** [review]
Looks fine.

**Review by Wil-Lee — APPROVED** [review]

**Review by ToastOfToastyness — APPROVED** [review]

File: src/main/java/com/peerrequest/app/services/EmailSender.java [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+     * of this sender. Needs to be called to create MimeMessage instances
+     * that can be prepared by the client and passed to send(MimeMessage).
+     *
+     * @return the new MimeMessage instance
+     * @see #send(MimeMessage)
+     * @see #send(MimeMessage[])
+     */
+    @Override
+    public MimeMessage createMimeMessage() {
+        return null;
```
Comment (bits0rcerer) [inline src/main/java/com/peerrequest/app/services/EmailSender.java:68]: Use `throw new RuntimeException("not implemented")` for stubs

**Pull request merged**